### ISSUE TITLE
Submission file API does not return genre information

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#7760]
**Describe the bug**
When requesting a submission file from the API, the `genre` property is always `null`. It should be an object like:

```json
{
  "id": 1,
  "dependent": false,
  "name": {
    "en_US": "example.docx"
  },
  "supplementary": false
}
```

**To Reproduce**
1. Go to the submission wizard
2. Go to step 2
3. Upload a file
4. Click "other" when asked about the file type and select any file type and save.
6. See that the file type doesn't update. If you inspect the XHR request in the Network tab of the dev tools, the `genre` property returned when the form was saved is `null`.

**What application are you using?**
 `main` branch (3.4  pre-release)

**Additional information**
I can reproduce this with a `GET` request to the API, like `api/v1/submissions/1/files/1`. I've checked the database and the `genreId` is set there. The relevant code is here:

https://github.com/pkp/pkp-lib/blob/ccb859831894e0a7a4b7ef34ad2b0083bcc826a9/classes/submissionFile/maps/Schema.inc.php#L137

I've checked this and the function is working. It returns the array. But somehow `genre` ends up `null` anyway.  This is the handler code that calls the map:

https://github.com/pkp/pkp-lib/blob/ccb859831894e0a7a4b7ef34ad2b0083bcc826a9/api/v1/submissions/PKPSubmissionFileHandler.inc.php#L246-L248

**Comment by NateWr** [comment]
Ok this turned out to be simple. Once fixed, it exposed an issue with the submission files list panel. It needed to be updated since an API change as part of #7265. These PRs should address both problems.

PRs:
https://github.com/pkp/pkp-lib/pull/7763
https://github.com/pkp/ui-library/pull/184

Tests:
https://github.com/pkp/omp/pull/1080 (OMP since it was OMP/PHP8.1 that had the original test failures)

**Comment by asmecher** [comment]
Interesting! Do you know why this was causing test failures only in a single combination of PHP and MySQL?

**Comment by NateWr** [comment]
No! I suspect there's a race condition of some kind that may be causing the test failure. I had to restart the PHP 8.0 test with this batch.

The race condition may be this line:

https://github.com/pkp/pkp-lib/blob/f1224bde174c4d07f8b9531b58243d5bc5ed52ae/cypress/support/commands.js#L206

The button to save the file genre is clicked but the test doesn't wait for confirmation that it is complete before moving forward. I'm not sure why the tests were ever passing, because the `should('not.exist')` element always remained. Maybe that line is not working properly or something.

The test failure seems to happen here:

https://github.com/pkp/pkp-lib/blob/f1224bde174c4d07f8b9531b58243d5bc5ed52ae/cypress/support/commands.js#L222

Which makes me think that the "save and continue to next step" part is not working. Probably a validation error is being returned. This may point to the same problem. If someone tries to save and continue to next step before the submission file's genre is set, I think it throws a validation error that says they need to select the file genre.

I got wrapped up in fixing this bug, so I didn't look further into it.

(ps - PRs merged so closing this)
